features/remotepollerng/daemon/src/main/java/org/opennms/netmgt/remotepollerng/RemotePollerd.java

```diff
@@ -196,6 +199,88 @@ public void scheduleAllServices() {
         });
     }
 
+    private Map<JobKey, RemotePolledService> getMapOfScheduledServices(final String locationName) {
+        final Map<JobKey, RemotePolledService> mapOfScheduledServices = new TreeMap<>();
+        try {
+            for(final JobKey jobKey : scheduler.getJobKeys(GroupMatcher.jobGroupEquals(locationName))) {
+                try {
+                    final JobDetail jobDetail = scheduler.getJobDetail(jobKey);
+
+                    if (locationName.equals(jobDetail.getJobDataMap().get(RemotePollJob.LOCATION_NAME)) &&
+                        jobDetail.getJobDataMap().get(RemotePollJob.REMOTE_POLLER_BACKEND) == this) {
+                        mapOfScheduledServices.put(jobKey, (RemotePolledService) jobDetail.getJobDataMap().get(RemotePollJob.POLLED_SERVICE));
+                    }
+                } catch (SchedulerException e) {
+                    LOG.warn("Failed to retrieve job details {}.", jobKey, e);
+                }
+            }
+        } catch (SchedulerException e) {
+            LOG.warn("Failed to query scheduled jobs.", e);
+        }
+
+        return mapOfScheduledServices;
+    }
+
+    public void handleConfigurationChanged() {
+        final Map<String, List<RemotePolledService>> servicesByPackage = new HashMap<>();
+
+        try {
+            this.pollerConfig.update();
+        } catch (IOException e) {
+            LOG.warn("Error reloading poller-configuration.xml");
+        }
+
+        LOG.info("Re-scheduling all services...");
+        sessionUtils.withReadOnlyTransaction(() -> {
+            for (final OnmsMonitoringLocation loc : monitoringLocationDao.findAll()) {
+                final List<String> pollingPackageNames = loc.getPollingPackageNames();
+                LOG.debug("Location '{}' has polling packages: {}", loc.getLocationName(), pollingPackageNames);
+
+                final Set<RemotePolledService> servicesToBeScheduled = new HashSet<>();
+
+                for (final String pollingPackageName : pollingPackageNames) {
```

Comment:
This is basically a duplicate of the code in {{scheduleAllServices}}. Can we deduplicate this?

Reply:
done